src/darker/tests/test_verification.py

```diff
@@ -30,3 +29,23 @@ def test_verify_ast_unchanged(src_content, dst_content, expect):
         assert expect is AssertionError
     else:
         assert expect is None
+
+
+def test_binary_search_premature_result():
+    """``darker.verification.BinarySearch``"""
+    with pytest.raises(RuntimeError):
+
+        _ = BinarySearch(0, 5).result
+
+
+def test_binary_search():
+    """``darker.verification.BinarySearch``"""
+    search = BinarySearch(0, 5)
+    tries = []
+    while not search.found:
+        tries.append(search.get_next())
+
+        search.respond(tries[-1] > 2)
+
+    assert search.result == 3
+    assert tries == [0, 3, 2]
```

Comment:
Sort-of fuzzy-testing
```suggestion
    assert tries == [0, 3, 2]


@pytest.mark.parametrize("i", range(50))
def test_binary_search_in_50(i):
    b = BinarySearch(0, 50)
    while not b.found:
        b.respond(b.get_next() >= i)
    assert b.result == i

```